vulnerabilities/tests/test_default_improver.py

```diff
@@ -0,0 +1,147 @@
+    """
+    Test that DefaultImprover.get_inferences() returns an empty list when given an advisory with no affected packages
+    """
+    advisory_data = AdvisoryData(
+        aliases=["CVE-2020-1234"],
+        summary="Test summary",
+        references=[
+            Reference(
+                url="https://www.example.com/",
+            )
+        ],
+        affected_packages=[],
+    )
+    default_improver = DefaultImprover()
+    expected_inference = Inference(
+        vulnerability_id=None,
+        aliases=["CVE-2020-1234"],
+        confidence=100,
+        summary="Test summary",
+        affected_purls=None,
+        fixed_purl=None,
+        references=[Reference(reference_id="", url="https://www.example.com/", severities=[])],
+    )
+    expected = [expected_inference]
+    result = list(default_improver.get_inferences(advisory_data))
+    assert result == expected
+
+
+def test_default_improver_with_affected_packages():
+    """
+    Test that DefaultImprover.get_inferences() returns an empty list when given an advisory with no affected packages
+    """
+    advisory_data = AdvisoryData(
+        aliases=["CVE-2020-1234"],
+        summary="Test summary",
+        references=[
+            Reference(
+                url="https://www.example.com/",
+            )
+        ],
+        affected_packages=[
+            AffectedPackage(
+                package=PackageURL(
+                    name="python",
+                    namespace="rpms",
+                    type="rpm",
+                ),
+                affected_version_range=None,
+                fixed_version="3.7.0",
+            )
+        ],
+    )
+    default_improver = DefaultImprover()
+    expected_inference = Inference(
+        vulnerability_id=None,
+        aliases=["CVE-2020-1234"],
+        confidence=100,
+        summary="Test summary",
+        affected_purls=[],
+        fixed_purl=PackageURL(
+            type="rpm",
+            namespace="rpms",
+            name="python",
+            version="3.7.0",
+            qualifiers={},
+            subpath=None,
+        ),
+        references=[Reference(reference_id="", url="https://www.example.com/", severities=[])],
+    )
+    result = list(default_improver.get_inferences(advisory_data))
+    expected = [expected_inference]
+    assert result == expected
+
+
+def test_default_improver_with_alpine(regen=VULNERABLECODE_REGEN_TEST_FIXTURES):
+    """
+    Test that DefaultImprover.get_inferences() returns an empty list when given an advisory with alpine advisories
+    """
+    default_improver = DefaultImprover()
+    response_file = os.path.join(TEST_DATA, f"alpine-input.json")
+    with open(response_file, "r") as f:
+        advisory_data = json.load(f)
+    expected_file = os.path.join(TEST_DATA, f"alpine-expected.json")
+    result = [
+        data.to_dict()
+        for data in list(default_improver.get_inferences(AdvisoryData.from_dict(advisory_data)))
+    ]
+    if regen:
```

Comment:
We will need to switch to this kind of approach IMHO, e.g. using commoncode plus this test helper: `util_tests.check_results_against_json()` here and below.
https://github.com/nexB/vulnerablecode/blob/d1f21983d39592b7470a1db038d66c0031053b30/vulnerabilities/tests/test_nginx.py#L232